### PR TITLE
Attempt to simplify glyf to_path

### PR DESCRIPTION
**Pull request by rsheeter**
I found the original code, directly ported from FreeType, a bit hard to follow and attempted to break it into smaller more tractable parts. Meant to make #818, which will rebase on top, simpler.

### CODE REVIEW

File: read-fonts/src/tables/glyf.rs [hunk 1]
```diff
@@ -665,7 +667,254 @@ impl fmt::Display for ToPathError {
...
+            });
+        }
+        Ok(Self {
+            base_offset,
+            points,
+            flags,
+        })
+    }
+
+    fn drain(self, sink: &mut impl Pen) -> Result<(), ToPathError> {
```
Comment (anthrotype) [inline read-fonts/src/tables/glyf.rs:901]: **¡bikeshed alert!** Draining a sink is not as nice a metaphor as drawing with a pen ... How about draw(pen) ? :)

Comment (rsheeter) [reply read-fonts/src/tables/glyf.rs:901]: I was trying to keep the sink naming @dfrg  had going alive. I tend to agree that draw(pen) better describes what's happening, will fix.

**Review by cmyr — APPROVED** [review]
No notes, but this is a very nice looking patch, and I think the logic is definitely easier to follow.

File: read-fonts/src/tables/glyf.rs (hunk 1 above)
Comment (rsheeter) [reply read-fonts/src/tables/glyf.rs]: Done

**Comment by rsheeter** [comment]
Confirmed good w/@dfrg, merging.

**Comment by dfrg** [comment]
yes, +1 from me. Nice work!
